metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -860,7 +886,7 @@ public void handleCommit(BatchReader<ApiMessageAndVersion> reader) {
                             int i = 1;
                             for (ApiMessageAndVersion message : messages) {
                                 try {
-                                    replay(message.message(), Optional.empty(), offset);
+                                    replay(message.message(), Optional.empty(), offset + i - 1, epoch);
```

Comment:
The `offset` -> `offset + i - 1` change, was that a bug previously? Were we just sending the batch offset to replay before?

Reply:
previously it was sending the end offset of the batch. it now sends the actual record offset.